plugin/src/main/kotlin/nl/avisi/anowalke/service/PsiService.kt

```diff
@@ -0,0 +1,22 @@
+package nl.avisi.anowalke.service
+
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiNameIdentifierOwner
+import com.intellij.psi.util.forEachDescendantOfType
+import nl.avisi.anowalke.toolWindow.FluffyListener
+
+class PsiService(fluffyListener: FluffyListener) {
+
+    private val log = Logger.getInstance(this::class.java.name)
+
+    fun extract(psiFile: PsiFile): List<String> {
+        val expressions = ArrayList<String>()
+
+        psiFile.forEachDescendantOfType<PsiNameIdentifierOwner> { it ->
+            it.name?.let { expressions.add(it) }
+        }
+        expressions.forEach { log.warn(it) }
+        return expressions
```

Comment:
Deze functie kan volgens mij makkelijker met `collectDescendantsOfType`.
Iets als:
````
fun extract(psiFile: PsiFile): List<String> =
    psiFile.collectDescendantsOfType<PsiNameIdentifierOwner> { it.name.isNotEmptyOrNull() }
        .also { log.warn(it) }
````

Reply:
collectDescendantsOfType verwacht een boolean maar krijgt een unit zegt hij. Of hij vraagt om het returntype te veranderen naar List<PsiNameIdentifierOwner>

Reply:
````
fun extract(psiFile: PsiFile): List<String> =
    psiFile.collectDescendantsOfType<PsiNameIdentifierOwner> { it.name != null }
        .map { it.name }
        .also { log.warn(it) }
````

Zoiets dan?